uplink/ecclient/client.go

```diff
@@ -148,6 +149,18 @@ func (ec *ecClient) Put(ctx context.Context, limits []*pb.AddressedOrderLimit, p
 	}()
 
 	successes := int(atomic.LoadInt32(&successfulCount))
+	mon.IntVal("segment_pieces").Observe(int64(len(limits)))
+	mon.FloatVal("segment_pieces_percent_success").Observe(float64(successes) / float64(rs.OptimalThreshold()))
+
+	var failures int
+	for range limits {
+		info := <-infos
+		if info.err != nil && err != context.Canceled {
```

Comment:
just curious what context canceled errors are reported as if not failures? are they expected/normal so don't need to be reported?

Reply:
Yea that was the thought. We're really only interested in stats on the pieces that either failed for some other reason or were successful. We have the total number of pieces so we can always calculate how many were cancelled if we need.

Reply:
We should probably use `errs2.IsCanceled(err)` from `storj.io/storj/internal/errs2`